relay-server/src/actors/processor.rs

```diff
@@ -1267,6 +1268,95 @@ impl EnvelopeProcessorService {
         Ok(())
     }
 
+    fn process_replays_combine_items(
+        &self,
+        state: &mut ProcessEnvelopeState,
+    ) -> Result<(), ProcessingError> {
+        let project_state = &state.project_state;
+        let combined_envelope_items =
+            project_state.has_feature(Feature::SessionReplayCombinedEnvelopeItems);
+
+        if combined_envelope_items {
+            // If this flag is enabled, combine both items into a single item,
+            // and remove the original items.
+            // The combined Item's payload is a MsgPack map with the keys
+            // "replay_event" and "replay_recording".
+            // The values are the original payloads of the items.
+            let envelope = &mut state.envelope_mut();
+            if let Some(replay_event_item) =
+                envelope.take_item_by(|item| item.ty() == &ItemType::ReplayEvent)
+            {
+                if let Some(replay_recording_item) =
+                    envelope.take_item_by(|item| item.ty() == &ItemType::ReplayRecording)
+                {
+                    // Split the headers from the body so they can be stored on separate keys.
+                    // This was previously done in another processing step (`process_replays`). To
+                    // be here we know it must have been successful otherwise the envelope would
+                    // have been dropped.
+                    //
+                    // TODO: Should we handle this possible error state more gracefully?
+                    // Potentially following the error pattern established in the msgpack
+                    // serialization step below?
+                    let (headers, body) =
+                        split_headers_from_body(replay_recording_item.payload().into())
+                            .expect("This is pre-validated and should not fail.");
```

Comment:
Instead of performing this split twice (once in `process_replays` and once here), I would unite the two functions and serialize either back into a recording item _or_ into a message pack item:

```mermaid
flowchart TD
    A[event, recording] -->|split| B[event, header, body]
    B -->|scrub| C[event, header, body]
    C --> D{combine?}
    D -->|yes| E[MsgPack: event, header, body]
    D -->|no| F[event, recording]
```

Reply:
@jjbayer Whats the best way to emit outcomes for dropped items?  The only way I've found to do it is with `retain_items` but it seems like that method won't work for this use case.

Reply:
You could manually call the function that `retain_items` calls: https://github.com/getsentry/relay/blob/c057e370834049843932c3d11ac4d74850aaeb1d/relay-server/src/utils/managed_envelope.rs#L231-L243

Reply:
@jjbayer I am out for a little while.  Josh Ferge will be re-taking over the PR so please direct feedback items to him.  Thank you.